crates/nu-command/src/commands/format/format_filesize.rs

```diff
@@ -71,12 +69,34 @@ async fn process_row(
     Ok({
         let replace_for = get_data_by_column_path(&input, &field, move |_, _, error| error);
         match replace_for {
-            Ok(s) => match convert_bytes_to_string_using_format(s, format) {
-                Ok(b) => OutputStream::one(ReturnSuccess::value(
-                    input.replace_data_at_column_path(&field, b).expect("Given that the existence check was already done, this shouldn't trigger never"),
-                )),
-                Err(e) => OutputStream::one(Err(e)),
-            },
+            Ok(s) => {
+                let byte_num = match s.value {
+                    Primitive(Filesize(s)) => {
+                        if let Some(value) = s.to_u128() {
```

Comment:
Err specifically this one.

But I guess maybe both? I was thinking you might want to check for the filesize you want closer to where it's used.

For the first question, I was wondering if we might want an `.is_filesize` so we don't need a full pattern match